src/Engines/Native/nativeShaderProcessors.ts

```diff
@@ -0,0 +1,24 @@
+import { Nullable } from '../../types';
+import { WebGL2ShaderProcessor } from '../WebGL/webGL2ShaderProcessors';
+import { ShaderProcessingContext } from '../Processors/shaderProcessingOptions';
+
+declare type ThinEngine = import("../thinEngine").ThinEngine;
+declare type NativeEngine = import("../nativeEngine").NativeEngine;
+
+/** @hidden */
+export class NativeShaderProcessor extends WebGL2ShaderProcessor {
+    public postProcessor(code: string, defines: string[], isFragment: boolean, processingContext: Nullable<ShaderProcessingContext>, engine: ThinEngine) {
+        code = super.postProcessor(code, defines, isFragment, processingContext, engine);
+
+        // Depending on API, depth range is [-1..1] or [0..1]
+        // This is defined by homogeneousDepth in bgfx caps
+        // Flip Y + convert z range from [-1,1] to [0,1]

```

Comment:
I think "Flip Y" should be removed in the comment, as the code is doing only the z range remapping.